pkg/deviceshifu/deviceshifubase/deviceshifubase_config.go

```diff
@@ -136,42 +136,47 @@ func NewDeviceShifuConfig(path string) (*DeviceShifuConfig, error) {
 
 // NewEdgeDevice new edgeDevice
 func NewEdgeDevice(edgeDeviceConfig *EdgeDeviceConfig) (*v1alpha1.EdgeDevice, *rest.RESTClient, error) {
-	var config *rest.Config
-	var err error
-
-	if edgeDeviceConfig.KubeconfigPath != "" {
-		config, err = clientcmd.BuildConfigFromFlags("", edgeDeviceConfig.KubeconfigPath)
-	} else {
-		config, err = rest.InClusterConfig()
-	}
-
+	config, err := getRestConfig(edgeDeviceConfig.KubeconfigPath)
 	if err != nil {
 		klog.Fatalf("Error parsing incluster/kubeconfig, error: %v", err.Error())
 		return nil, nil, err
 	}
 
-	client, err := NewEdgeDeviceRestClient(config)
+	client, err := newEdgeDeviceRestClient(config)
 	if err != nil {
 		klog.Fatalf("Error creating EdgeDevice custom REST client, error: %v", err.Error())
 		return nil, nil, err
 	}
+	device, err := createDevice(client, edgeDeviceConfig)
```

Comment:
Not sure if it is necessary to put this into a function, since it is only used here.